client/src/components/river/index.tsx

```diff
@@ -155,11 +206,60 @@ const FhirRiverView = (): React.ReactElement => {
             disabled={!selectedSource.id || credentialsMissing}
             loading={running}
             className="button-submit"
-            onClick={onClickRun}
+            onClick={onClickCreateBatch}
           >
             Make the river flow
           </Button>
         </div>
+        <h1>Cancel a batch</h1>
+        <StringSelect
+          items={
+            batchList.data
+              ? Object.keys(batchList.data).map(
+                  (batchId: string) => batchList[batchId].timestamp
+                )
+              : []
```

Comment:
What do you think about destructing the query response in the action? I don't know what people usually do I think it would be nice to avoid this logic here and put it closer to the call

Reply:
The action should stay minimal https://redux.js.org/style-guide/style-guide#put-as-much-logic-as-possible-in-reducers . Also the reducer should store normalized data in the global state and handle errors as well. What I can do is add a selector.

Reply:
Oh right, maybe it's the selector's job to do that. What do you think?